### PR TITLE
Asset screen implementation

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The screen that shows a list of assets with their information has been implemented.

Video:
https://user-images.githubusercontent.com/84338513/150985502-58977b8a-6977-4d74-b818-0ec2a38d620b.mov

Unit test passed:
Screen Shot 2022-01-25 at 17 54 14
Screen Shot 2022-01-25 at 17 54 25

Coverage:
Screen Shot 2022-01-25 at 17 54 41

### CODE REVIEW

**Comment by ezequielsetzesglobant** [comment]
The branch will be closed due to problems with it when doing a rebase. And the link to the new pr will be left.

File: stockApp/stockApp/Data/Entity/Asset.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+struct Asset: Codable {
+
+    let barcode: String
+    let assetName: String
+    let userMail: String
+    let recordDateTime: String
+    let site: String
+
+    enum CodingKeys: String, CodingKey {
+        case barcode = "barcode"
```
Comment (fn1983) [inline stockApp/stockApp/Data/Entity/Asset.swift:12]:
```suggestion
        case barcode
```

File: stockApp/stockApp/Data/Entity/Asset.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        case assetName = "asset_name"
+        case userMail = "user_mail"
+        case recordDateTime = "record_date_time"
+        case site = "site"
```
Comment (fn1983) [inline stockApp/stockApp/Data/Entity/Asset.swift:16]:
```suggestion
        case site
```

File: stockApp/stockApp/Data/Entity/Fields.swift [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Foundation
+
+struct Fields: Codable {
+
+    let values: [String] = ["Barcode", "Asset Name", "User Mail", "Location"]
```
Comment (fn1983) [inline stockApp/stockApp/Data/Entity/Fields.swift:5]: you can have something like this

```
enum FieldsType: String, CaseIterable {
        case barcode = "Barcode" 
        ....
}

let values: [FieldsType] = FieldsType.allCases
```

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+    // MARK: - Properties
+
+    let backView: UIView = {
+        let view = UIView()
+        view.backgroundColor = UIColor.black.withAlphaComponent(CGFloat(Constants.backViewAlpha))
+        return view
+    }()
+
+    let barcodeUILabel: UILabel = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:43]:
```suggestion
    let barcodeLabel: UILabel = {
```

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        label.font = UIFont(name: Strings.heeboFont, size: CGFloat(Constants.fontSize))
+        label.font = label.font.withSize(CGFloat(Constants.fontSizeLabel))
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = Constants.numberOfLines
+        label.textColor = Color.porcelainWhite
+        label.textAlignment = .left
+        return label
+    }()
+
+    let assetNameUILabel: UILabel = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:54]:
```suggestion
    let assetNameLabel: UILabel = {
```

we should add types in variable names.. if you change to another type.. like CUstomLabel. the variable name will contain UILabel .

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        label.font = UIFont(name: Strings.heeboFont, size: CGFloat(Constants.fontSize))
+        label.font = label.font.withSize(CGFloat(Constants.fontSizeLabel))
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = Constants.numberOfLines
+        label.textColor = Color.porcelainWhite
+        label.textAlignment = .left
+        return label
+    }()
+
+    let globerMailUILabel: UILabel = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:65]:
```suggestion
    let globerMailLabel: UILabel = {
```

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        label.font = UIFont(name: Strings.heeboFont, size: CGFloat(Constants.fontSize))
+        label.font = label.font.withSize(CGFloat(Constants.fontSizeLabel))
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = Constants.numberOfLines
+        label.textColor = Color.porcelainWhite
+        label.textAlignment = .left
+        return label
+    }()
+
+    let siteUILabel: UILabel = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:76]:
```suggestion
    let siteLabel: UILabel = {
```

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+        label.font = UIFont(name: Strings.heeboFont, size: CGFloat(Constants.fontSize))
+        label.font = label.font.withSize(CGFloat(Constants.fontSizeLabel))
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = Constants.numberOfLines
+        label.textColor = Color.porcelainWhite
+        label.textAlignment = .left
+        return label
+    }()
+
+    let imageUIImageView: UIImageView = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:87]:
```suggestion
    let arrowImageView: UIImageView = {
```
give context to the name!

File: stockApp/stockApp/Presentation/View/AssetView.swift [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    static let defaultTrailing: Int = 0
+}
+
+// MARK: - Class
+
+class AssetView: UIView {
+
+    // MARK: - Properties
+
+    let searchUIButton: UIButton = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetView.swift:40]:
```suggestion
    let searchButton: UIButton = {
```

change others please!

File: stockApp/stockApp/Presentation/View/AssetView.swift [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        gradient.colors = [startColor.cgColor, endColor.cgColor]
+        gradient.zPosition = CGFloat(Ints.gradientZPosition)
+        layer.addSublayer(gradient)
+    }
+
+    func addActions() {
+        searchUIButton.addTarget(self, action: #selector(self.searchButtonTapped), for: .touchUpInside)
+    }
+}
+@objc extension AssetView {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetView.swift:170]:
```suggestion

@objc extension AssetView {
```

File: stockApp/stockApp/Presentation/View/AssetViewController.swift [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        self.navigationController?.navigationBar.barTintColor = Color.atlantisGreen
+        viewModel.getAssets(filter: "Location", value: "Tandil")
+    }
+
+    func setUpViewModel() {
+        viewModel.onFetchAssets = { [weak self] assets in
+            self?.assets = assets
+            self?.displayAssets()
+        }
+        viewModel.onNotAssetsFound = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetViewController.swift:50]:
```suggestion
        viewModel.onNotAssetsFound = { [weak self] in
        guard let self = self else { return }
```

File: stockApp/stockApp/Presentation/View/AssetViewController.swift [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+            self.displayNotAssetsFoundMessage()
+        }
+        viewModel.onThereIsNoInternet = {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetViewController.swift:53]:
```suggestion
        viewModel.onThereIsNoInternet = { [weak self] in
                guard let self = self else { return }
```

File: stockApp/stockApp/Presentation/View/AssetViewController.swift [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return assets.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Strings.cellIdentifier,
+                                                       for: indexPath) as? AssetTableViewCell else {
+            return AssetTableViewCell()
+        }
+        cell.setFields(assetViewData: assets[indexPath.row])
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetViewController.swift:93]:
```suggestion
        let assetViewData = assets[indexPath.row]
        cell.setFields(assetViewData: assetViewData)
```

File: stockApp/stockApp/Presentation/View/MainViewController.swift [hunk 14]
```diff
@@ -58,6 +58,7 @@ extension MainViewController: MainViewDelegate {
             coordinator?.goToCameraViewController()}
 
 	func searchButtonPressed(from view: MainView) {
-		viewModel.showSearchAlert()
+        // viewModel.showSearchAlert()
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/MainViewController.swift:61]:
```suggestion
```

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift [hunk 15]
```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+// MARK: - Class
+
+class AssetViewModel: AssetViewModelProtocol {
+
+    internal let useCase: GetAssetUseCaseProtocol
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift:7]:
```suggestion
    let useCase: GetAssetUseCaseProtocol
```

if yo don't have frameworks, always it's going to be internal..

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+    var onFetchAssets: (([AssetViewData]) -> Void)?
+    var onNotAssetsFound: (() -> Void)?
+    var onThereIsNoInternet: (() -> Void)?
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift:10]:
```suggestion
    var onNoInternetConnection: (() -> Void)?
```

check language ..

**Review by fn1983 — CHANGES_REQUESTED** [review]

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 5 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift:54]: Do you wanted to put that we shouldn't put a type, right?

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift (hunk 16 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift]: done

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift (hunk 15 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift]: done

File: stockApp/stockApp/Presentation/View/MainViewController.swift (hunk 14 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/MainViewController.swift]: done

File: stockApp/stockApp/Presentation/View/AssetViewController.swift (hunk 13 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetViewController.swift]: done

File: stockApp/stockApp/Presentation/View/AssetViewController.swift (hunk 12 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetViewController.swift]: done

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 8 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift]: done

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 7 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift]: done

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 6 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift]: done

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 5 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift]: done

File: stockApp/stockApp/Presentation/View/AssetTableViewCell.swift (hunk 4 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetTableViewCell.swift]: done

File: stockApp/stockApp/Data/Entity/Fields.swift (hunk 3 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Data/Entity/Fields.swift]: done

File: stockApp/stockApp/Data/Entity/Asset.swift (hunk 2 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Data/Entity/Asset.swift]: done

File: stockApp/stockApp/Data/Entity/Asset.swift (hunk 1 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Data/Entity/Asset.swift]: done

**Review by fn1983 — APPROVED** [review]
LGTM .. there are some things to polish but.. so far so good

**Review by AgustinaB-Globant — APPROVED** [review]

**Pull request merged**